_posts/2019-08-13-how-to-deploy-production-grade-vpc-aws.adoc

```diff
@@ -746,6 +756,9 @@ file for reference.
 [[test_wrapper_module]]
 ==== Test your wrapper module
 
+# TODO: again, this is not good advice. we need to mention they should only use `plan`, not `apply` for testing, and give them a much better
+# idea of exactly what they need to set. I think they should just create the terragrunt.hcl and run plan against that, which is explained further on,
+# So we should have them create the terragrunt.hcl, then run plan, git tag/push, and then run apply.
```

Comment:
Actually, for the VPC, we want to encourage apply based testing because you can do that without causing account wide damage. The Landing Zone modules are special in that it works off of cross account and account wide resources, which makes it really hard to do any kind of `apply` based testing. Since the VPC module only creates resources in a single account, it is much easier to do apply based testing because you can isolate the resources in a sandbox or testing account.

What this is making me realize is that you can't really use any of these guides in isolation. We really need an opinionated, structured guide that details the patterns we follow and intend to take (including how to do automated testing), and introduce them to those patterns before diving into something like this. Otherwise, none of the things we wrote here make much sense, and it is too easy to shoot yourself in the foot by following blindly.

Reply:
> can't really use any of these guides in isolation
I agree. And also, if we operate from this assumption, we can write them with less duplication and more consistency across the guides.

> for the VPC, we want to encourage apply based testing
Got it! Makes sense.